warp/protocol-land/actions/issues.ts

```diff
@@ -284,6 +292,50 @@ export async function addCommentToIssue(
   return { state }
 }
 
+export async function updateIssueComment(
+  state: ContractState,
+  { caller, input: { payload } }: RepositoryAction
+): Promise<ContractResult<ContractState>> {
+  if (
+    isInvalidInput(payload, 'object') ||
+    isInvalidInput(payload.repoId, 'uuid') ||
+    isInvalidInput(payload.issueId, ['number', 'string']) ||
+    isInvalidInput(payload.comment, 'object') ||
+    isInvalidInput(payload.comment.id, 'number') ||
+    isInvalidInput(payload.comment.description, 'string')
+  ) {
+    throw new ContractError('Invalid inputs supplied.')
+  }
+
+  const repo = state.repos[payload.repoId]
+
+  if (!repo) {
+    throw new ContractError('Repo not found.')
+  }
+
+  const hasPermissions = caller === repo.owner || repo.contributors.indexOf(caller) > -1
```

Comment:
I think this will allow owner to edit comments of contributor, or vice versa

Reply:
Yes, I did so because we can edit comments of contributors or vice versa in GH too.

Reply:
@7i7o Thoughts here ?

Reply:
I think we need to keep in check on the features we want to add. Not every feature will be useful to users at this stage and showing history would add additional network calls to the page otherwise unwanted complexity, specially when we dont have dedicated servers like github does. Focusing on 80-20 rule is the way to go.